react/features/base/react/components/web/Message.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import React, { Component } from 'react';
+import { toArray } from 'react-emoji-render';
+
+import Linkify from './Linkify';
+
+type Props = {
+
+    /**
+     * The body of the message.
+     */
+    text: string
+};
+
+/**
+ * Renders the content of a chat message.
+ */
+class Message extends Component<Props> {
+    /**
+     * Initializes a new {@code Message} instance.
+     *
+     * @param {Props} props - The props of the component.
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        // Bind event handlers so they are only bound once for every instance
+        this._processMessage = this._processMessage.bind(this);
+    }
+
+    /**
+     * Parses and builds the message tokens to include emojis and urls.
+     *
+     * @returns {ReactElement}
```

Comment:
returns Array<string> ?

Reply:
Well, it's wrapped in a fragment so I thought it would be safe to call it a ReactElement 🤔. Should I simplify the render return as well?

Reply:
I think render actually returns a reactElement, , but this function call is wrapped in { .. }  inside the fragment so that does not make it mandatory a React Component, can be plain text can be anything. In our case its an array of strings

Reply:
Oh, sorry, I thought we're discussing the render jsdoc. Yes, I'll change that.